slsDetectorServers/eigerDetectorServer/Beb.c

```diff
@@ -298,125 +297,146 @@ int Beb_IsTransmitting(int *retval, int tengiga, int waitForDelay) {
+        value |= BEB_CONFIG_TOP_MSK;
+        break;
+    case OW_BOTTOM:
+        value |= BEB_CONFIG_OW_TOP_MSK;
+        value &= ~BEB_CONFIG_TOP_MSK;
+        break;
+    default:
+        LOG(logERROR, ("Unknown top index in Beb: %d\n", ind));
         Beb_close(fd, csp0base);
+        return 0;
+    }
 
-    return ret;
+    char *top_names[] = {TOP_NAMES};
+    int newval = Beb_Write32(csp0base, BEB_CONFIG_WR_OFST, value);
+    if (newval != value) {
+        LOG(logERROR,
+            ("Could not set Top flag to %s in Beb\n", top_names[ind]));
+        Beb_close(fd, csp0base);
+        return 0;
+    }
+    LOG(logINFOBLUE,
+        ("%s Top flag to %s in Beb\n",
+         (ind == TOP_HARDWARE ? "Resetting" : "Overwriting"), top_names[ind]));
+    Beb_close(fd, csp0base);
+    return 1;
 }
 
-/* do not work at the moment */
-int Beb_SetSlaveViaSoftware() {
-
+int Beb_SetMaster(enum MASTERINDEX ind) {
     if (!Beb_activated)
         return 0;
 
-    // mapping new memory
     u_int32_t *csp0base = 0;
-    u_int32_t value = 0, ret = 1;
-
-    // open file pointer
+    u_int32_t value = 0;
     int fd = Beb_open(&csp0base, XPAR_PLB_GPIO_SYS_BASEADDR);
     if (fd < 0) {
-        LOG(logERROR, ("Set Slave FAIL\n"));
-    } else {
-        value = Beb_Read32(csp0base, MASTERCONFIG_OFFSET);
-        value &= ~MASTER_BIT;
-        value |= OVERWRITE_HARDWARE_BIT;
-        int newval = Beb_Write32(csp0base, MASTERCONFIG_OFFSET, value);
-        if (newval != value) {
-            LOG(logERROR, ("Could not set Slave via Software\n"));
-        } else {
-            ret = 0;
-        }
+        LOG(logERROR, ("Set Master FAIL, could not open fd in Beb\n"));
+        return 0;
+    }
+    value = Beb_Read32(csp0base, BEB_CONFIG_WR_OFST);
+    switch (ind) {
+    case MASTER_HARDWARE:
+        value &= ~BEB_CONFIG_OW_MASTER_MSK;
+        break;
+    case OW_MASTER:
+        value |= BEB_CONFIG_OW_MASTER_MSK;
+        value |= BEB_CONFIG_MASTER_MSK;
+        break;
+    case OW_SLAVE:
+        value |= BEB_CONFIG_OW_MASTER_MSK;
+        value &= ~BEB_CONFIG_MASTER_MSK;
+        break;
+    default:
+        LOG(logERROR, ("Unknown master index in Beb: %d\n", ind));
+        Beb_close(fd, csp0base);
+        return 0;
     }
 
-    // close file pointer
-    if (fd > 0)
+    char *master_names[] = {MASTER_NAMES};
+    int newval = Beb_Write32(csp0base, BEB_CONFIG_WR_OFST, value);
+    if (newval != value) {
+        LOG(logERROR,
+            ("Could not set Master flag to %s in Beb\n", master_names[ind]));
         Beb_close(fd, csp0base);
+        return 0;
+    }
+    LOG(logINFOBLUE, ("%s Master flag to %s in Beb\n",
+                      (ind == MASTER_HARDWARE ? "Resetting" : "Overwriting"),
+                      master_names[ind]));
 
-    return ret;
+    Beb_close(fd, csp0base);
+    return 1;
 }
 
-int Beb_Activate(int enable) {
-    // mapping new memory
+int Beb_SetActivate(int enable) {
+    enable = enable == 0 ? 0 : 1;
```

Comment:
I had to think a bit before I understood what and why =)
I would move the statement to just before assigning to Beb_activated an maybe replace either with the 
short but cryptical:

`Beb_activated = !!enable;`

or the more readable:
`if (enable)
   enable = 1;
Beb_activated = enable;`

Reply:
Not one liners... but the editor doesn't like me =)

Reply:
I did a if (enable < 0 ) print error and return 0. Otherwise, we dont care if enable > 1. I removed the line enable = enable == 0... from Beb.c and list.c